homeassistant/components/media_player/yamaha.py

```diff
@@ -50,7 +50,10 @@ def update(self):
         self._muted = self._receiver.mute
         self._volume = (self._receiver.volume/100) + 1
         self._current_source = self._receiver.input
-        self._source_list = list(self._receiver.inputs().keys())
+
+        if self.source_list is None:
```

Comment:
No, the sources are fixed and once received we do not need to alter them anymore